lib/output.js

```diff
@@ -354,7 +356,18 @@ function OutputStream(options) {
         force_semicolon : force_semicolon,
         to_ascii        : to_ascii,
         print_name      : function(name) { print(make_name(name)) },
-        print_string    : function(str, quote) { print(encode_string(str, quote)) },
+        print_string    : function(str, quote, escape_directive) {
+            var encoded = encode_string(str, quote);
+            if (escape_directive === true && encoded.indexOf("\\") === -1) {
```

Comment:
As a matter of style, one generally doesn't compare to exact true or exact false unless the variable is overloaded to also be an object or a number. At least that's the case in the rest of the Uglify code base.